DogsList/DogsList/Controllers/SearchHistoryViewController.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  SearchHistoryViewController.swift
+//  DogsList
+//
+//  Created by Nick Krasnitskiy on 16.07.2021.
+//
+
+import UIKit
+
+class SearchHistoryViewController: UIViewController {
+    
+    enum Section: CaseIterable {
+        case main
+    }
+    
+    let searchHistory: [String]
+    
+    init(with searchHistory: [String]) {
+        self.searchHistory = searchHistory
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    typealias DataSource = UICollectionViewDiffableDataSource<Section, String>
+    private var dataSource: DataSource!
```

Comment:
do not use force unwrap for optionals

Reply:
I thought it's an Apple standard practice from its samples.

Reply:
no, it's used in samples for brevity only.